### PR TITLE
MockWatcher.trade and MockClient

### PR DESCRIPTION
**Pull request by sebastianst**
- utils/arrays: Add equalArray function to new package utils/arrays
- ledger: Use equalArray in Address.equals
- test: Allow optional TradeOffer in newRandomTrade
- test: Add MockWatcher.trade to inject Trade transactions
- !fixup
- test: Add MockClient (extends MockWatcher)

### CODE REVIEW

File: src/utils/arrays/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+
+export type TypedArray = Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array;
+
+export function equalArray(x: TypedArray, y: TypedArray): boolean {
+	return (x.length == y.length)
```
Comment (ndzik) [inline src/utils/arrays/index.ts:6]:
```suggestion
	return (x.length === y.length)
```

File: src/utils/arrays/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+		&& (x.every((a, i) => y[i] == a));
```
Comment (ndzik) [inline src/utils/arrays/index.ts:7]: Considering the function signature this is correct, because `x` and `y` can be of different `TypedArray` types, which would require a comparison function with type coercion.

If this was not intentional I would change the implementation to

```typescript
export functino equalArray<T extends TypedArray>(x: T, y: T): boolean { ... }
```

And 
```suggestion
		&& (x.every((a, i) => y[i] === a));
```

I did not test it, but I would not be surprised if comparing certain edge cases of `Int8Array` to `Uint8Array` and similar give unexpected results.

**Review by ndzik — CHANGES_REQUESTED** [review]

File: src/utils/arrays/index.ts (hunk 1 above)
Comment (sebastianst) [reply src/utils/arrays/index.ts:6]: I really need to get used to this `===` business in JS ^^

File: src/utils/arrays/index.ts (hunk 2 above)
Comment (sebastianst) [reply src/utils/arrays/index.ts:7]: I really need to get used to this `===` business in JS ^^

Comment (sebastianst) [reply src/utils/arrays/index.ts:7]: The nicest way would be to define an interface with property `length`, function `every` and `[]` index access. But I cannot be bothered right now to find out how this is done correctly.

**Review by ndzik — APPROVED** [review]
Absolute unit :mountain:
